### PR TITLE
Mention Curry dependency in README

### LINKED ISSUE #270: Explicitly mention Curry is external in README
The README code sample uses `curry` but the `curry` functions do not come with Argo. They are part of another library, [Curry](https://github.com/thoughtbot/Curry). We should reference this in the README to clarify that this will require an extra dependency.

relevant: #269

### PR DESCRIPTION
**Pull request by gilesvangruisen**
fixes #270

Perhaps it would be worthwhile to show an example of a curried `create` method in case people don't want to add an additional dependency?

### CODE REVIEW

**Comment by tonyd256** [comment]
I was thinking it might get more visibility being either before the example or right after the install guide

**Comment by gilesvangruisen** [comment]
Yeah I had it above the example initially. Happy to move it back there.

File: README.md [hunk 1]
```diff
@@ -100,6 +100,8 @@ if let j: AnyObject = json {
 
 For more information, see the [Documentation](Documentation/)
 
+Please note: the above example requires an additional, external module named [Curry](https://github.com/thoughtbot/Curry) to automatically curry `User.init`.
```
Comment (tonyd256) [inline README.md]: also it's not automatic so we should say "to use the `curry` function to curry `User.init`." or something like that

Comment (gilesvangruisen) [inline README.md]: True, "automatic" was a poor choice of wording. I'll make those changes once I'm back at my computer :+1:

**Comment by gilesvangruisen** [comment]
@tonyd256 thoughts on providing an example of a curried `create` method?

File: README.md [hunk 2]
```diff
@@ -100,6 +102,7 @@ if let j: AnyObject = json {
 
 For more information, see the [Documentation](Documentation/)
 
+
```
Comment (tonyd256) [inline README.md]: extra line?

**Comment by tonyd256** [comment]
I think an example of a curried create is something we want to have in our documentation. Then we could have a link in that blurb that pointed to the appropriate docs

**Comment by gilesvangruisen** [comment]
Good call, I think that makes more sense than adding clutter to the README

**Pull request merged**